### ISSUE TITLE
helix panics if cwd is removed

### ISSUE THREAD

**Issue by sigmaSd** [helix-editor/helix#8068]
### Summary

helix panics if cwd is removed

### Reproduction Steps

- mkdir a
- cd a
- from another terminal rm -rf a
- hx

expected: no crash
actual: panics
```
thread 'main' panicked at 'Couldn't determine current working directory: Os { code: 2, kind: NotFound, message: "No such file or directory" }', helix-loader/src/lib.rs:29:10
```

https://github.com/helix-editor/helix/pull/7185 says this is an internal invariant but in reality its used in many places accessible from the outside

### Platform

linux

### Terminal Emulator

weterm

### Helix Version

helix 23.05 (c9694f68)

**Comment by pascalkuthe** [comment]
I just don't see how this would work. If the cwd doesn't exists before helix is started then it's impossible for us ti the determine the cwd oath so all commands that use the cwd (:open, anytime you start an LSP, file picker, global search,...) just won't work properly.  I think its entirely reasonable to assume that the cwd exists at startup.


Especially because helix crashes immidietly (so no lost work) with an entirely reasonable error message I just don't think it's worth the extra headache to fix.

**Comment by cronhan** [comment]
I also recently experienced it, and even though I understood the message, it was not apparent what was wrong immediately since the shell did not give a similar error message (and my prompt includes the full CWD so even more "confusing"). So I think it might be good to at least _adjust_ the error message to also include the path that it tried but does not exist? Or add a hint to the message that the user should check whether that directory exists/was removed by another process?

**Comment by eminence** [comment]
I think the only other possible option is for helix to change to a different directory, when the cwd doesn't exist.  But then you're left with the question of what directory should that be, and then the subsequent possible confusion from users when that happens.   So the current behavior of refusing to start seems well justified.

I think probably a lot of people have come to associate panic error messages with bugs, rightly or wrongly.

Would you accept a PR that improves this message a bit?   Something like:

```
Error: Couldn't determine current working directory.  Check that '/tmp/a` exists.
```

**Comment by pascalkuthe** [comment]
Yeah I think helix should just print an error message and exit with an error code.
